### ISSUE TITLE
localhost proxy resolved to IPv6 (in version 2.3.1)

### ISSUE THREAD

**Issue by CheViana** [aio-libs/aiohttp#2402]
## Long story short

I'm using `aiohttp` to make calls via proxy "http://localhost:9999". "etc/hosts" maps localhost to IPv4 and IPv6 addresses.
`aiohttp` version 2.2.5 resolves that proxy value to "127.0.0.1:9999", `aiohttp` version 2.3.1 - to value "::1:9999".
I can't pinpoint changes in 2.3.1 that caused this behavior.
I found a workaround, but I think IPv4 should be used by default in this case.

## Expected behavior

Proxy "http://localhost:9999" got resolved to "127.0.0.1:9999" (IPv4 is default).

## Actual behavior

Proxy "http://localhost:9999" got resolved to "::1:9999" (IPv6 is used for some reason).

## Steps to reproduce

```
    async with aiohttp.ClientSession() as session:
            async with session.get(
                'https://github.com/aio-libs/aiohttp',
                proxy='http://localhost:9999'
            ) as response:
                response.raise_for_status()
                result = await response.read()
                return result
```

As far as I nailed it, line 785 in `connector.py` does host resolution:
`hosts = yield from self._resolve_host(req.url.raw_host, req.port)`

In version 2.2.5 `hosts[0]` equals:
```
{'hostname': 'localhost',
 'host': '127.0.0.1',
 'port': 9999,
 'family': <AddressFamily.AF_INET: 2>,
 'proto': 6,
 'flags': <AddressInfo.AI_NUMERICHOST: 4>}
```
In version 2.3.1 `hosts[0]` equals:
```
{'family': <AddressFamily.AF_INET6: 10>,
 'flags': <AddressInfo.AI_NUMERICHOST: 4>,
 'host': '::1',
 'hostname': 'localhost',
 'port': 9999,
 'proto': 6}
```
(`aiodns` is not used)

I've tried to call directly `socket.getaddrinfo` (AFAIK that is the function that does resolution under the hood):

```
>>> import socket
>>> socket.getaddrinfo('localhost', 9999, type=socket.SOCK_STREAM, family=socket.AF_INET)
[(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 4140)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 4140))]
>>> socket.getaddrinfo('localhost', 9999, type=socket.SOCK_STREAM, family=socket.AF_INET6)
[(<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('::1', 4140, 0, 0))]
```
`family` parameter is the trick, but I'm not modifying it in request code and I don't see where in `aiohttp` code it's value changed...

## Your environment

Linux. Python 3.6.
Contents of /etc/hosts:
```
127.0.0.1 localhost ...
::1 localhost ...
```
(`aiodns` is not used)

## Workaround

Specify socket family explicitly:

```
    connector = aiohttp.TCPConnector(family=socket.AF_INET)
    async with aiohttp.ClientSession(connector=connector) as session:
            async with session.get(
                'https://github.com/aio-libs/aiohttp',
                proxy='http://localhost:9999'
            ) as response:
                response.raise_for_status()
                result = await response.read()
                return result
```

**Comment by asvetlov** [comment]
I pretty sure aiohttp has no changes for it.
What is `hosts` list content? Not only `hosts[0]` but the whole list?

**Comment by johnlockwood-wf** [comment]
I hit this issue too

**Comment by asvetlov** [comment]
The issue should be fixed by #2424 
aiohttp will connect to all resolved IPs: IPv4 and IPv6 should be processed in parallel.

**Comment by asvetlov** [comment]
Fixed by #2447

**Comment by johnlockwood-wf** [comment]
That did indeed fix it for me. Thanks!
